File: src/main/java/io/github/zhengchalei/MyLivenessCheck.java
package io.github.zhengchalei;

import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.HealthCheckResponse;
import org.eclipse.microprofile.health.Liveness;

/**
 * 健康检查
 *
 * @author 郑查磊
 */
@Liveness
public class MyLivenessCheck implements HealthCheck {

    @Override
    public HealthCheckResponse call() {
        return HealthCheckResponse.up("alive");
    }

}

File: src/main/kotlin/io/github/zhengchalei/MyLivenessCheck.kt
package io.github.zhengchalei

import org.eclipse.microprofile.health.HealthCheck
import org.eclipse.microprofile.health.HealthCheckResponse
import org.eclipse.microprofile.health.Liveness

/**
 * 健康检查
 *
 * @author 郑查磊
 */
@Liveness
class MyLivenessCheck : HealthCheck {
    override fun call(): HealthCheckResponse {
        return HealthCheckResponse.up("alive")
    }
}
